### DISCUSSION TITLE
Can a DuckDB extension provide a custom physical operator?

### DISCUSSION THREAD

**Discussion by mwiewior** [duckdb/duckdb#9646] [Q&A] [answered]
Hi - I'm analysing the source code and documentation and trying to figure out if there is an easy way of implementing let's say custom (domain-specific) join strategy and plug in it using an extension mechanism? Most extensions I found are either providing additional SQL functions or custom data formats - is there any extension that implements custom physical operators that I could use as an inspiration?
Thanks!

**Comment by Tishj** [comment, marked as answer]
Yes, I believe Motherduck does this

I don't think there's an example available, I suggest browsing this repository.

Result collectors are also physical operators, those can be set on the connection with a C++ method, that's an easier path to start

**Reply by Tishj** [reply]
I think that requires making a parser extension + a extension statement, so when CreatePlan is called you can insert your own operator into the plan

**Reply by mwiewior** [reply]
Thanks a lot @Tishj !  - I will check this option.
